c++/src/kj/async-inl.h

```diff
@@ -2126,8 +2126,24 @@ class Coroutine<T>::Awaiter {
     return U(kj::mv(*value));
   }
 
-  void await_suspend(Coroutine::Handle coroutine) {
-    coroutine.promise().awaitSuspend(PromiseNode::from(promise), result);
+  bool await_suspend(Coroutine::Handle coroutine) {
+    auto& node = PromiseNode::from(promise);
+    auto& coroutineEvent = coroutine.promise();
+
+  // TODO(now): Should we call node.setSelfPointer() here?
```

Comment:
@kentonv I noticed most other places call setSelfPointer() immediately before onReady(). Should we do that here?